Source/SwiftLintFramework/Rules/LegacyConstructorRule.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  LegacyConstructorRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 29/11/15.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+import SwiftXPC
+
+public struct LegacyConstructorRule: Rule {
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "legacy_constructor",
+        name: "Legacy Constructor",
+        description: "Swift constructors are preferred over legacy convenience functions.",
+        nonTriggeringExamples: [
+            "CGPoint(x: 10, y: 10)",
+            "CGSize(width: 10, height: 10)",
+            "CGRect(x: 0, y: 0, width: 10, height: 10)",
+            "CGVector(dx: 10, dy: 10)",
+            "NSRange(location: 10, length: 1)",
+        ],
+        triggeringExamples: [
+            "CGPointMake(10, 10)",
+            "CGSizeMake(10, 10)",
+            "CGRectMake(0, 0, 10, 10)",
+            "CGVectorMake(10, 10)",
+            "NSMakeRange(10, 1)",
+        ]
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
```

Comment:
I have a few suggestions:

1. It's more performant to only scan the file for matches once, which can be done by merging all the regular expressions into one.
2. Instead of excluding comments and strings, we should check if the match is a `SyntaxKind.Identifier` which is more semantically correct, even though it probably won't make a difference in practice.

So this function could be refactored to this:

```swift
public func validateFile(file: File) -> [StyleViolation] {
    let constructors = ["CGRectMake", "CGPointMake", "CGSizeMake", "CGVectorMake",
        "NSMakeRange"]

    let pattern = "\\b(" + constructors.joinWithSeparator("|") + ")\\b"

    return file.matchPattern(pattern, withSyntaxKinds: [.Identifier]).map {
        StyleViolation(ruleDescription: self.dynamicType.description,
            location: Location(file: file, offset: $0.location))
    }
}
```